src/winrpath.cxx

```diff
@@ -598,44 +658,7 @@ bool CoffParser::NormalizeName(std::string &name)
                     DWORD section_data_start_offset = std::streamoff(mem.offset) + sizeof(IMAGE_ARCHIVE_MEMBER_HEADER) + psec_header->PointerToRawData;
                     // section start is longmember section pointer + index
                     char * section = *(mem.member->long_member->section_data+j);
-                    // Make a copy of the pointer
-                    char * section_search_start = &*section;
-                    char * search_terminator = section+data_size;
-                    // search section data for full name
-                    int name_len = name.size();
-                    while(section_search_start && (section_search_start < search_terminator)) {
-                        section_search_start = findstr(section_search_start, name.c_str(), data_size);
-                        if (section_search_start) {
-                            // we found a name, rename
-                            ptrdiff_t section_data_offset = section_search_start - section;
-                            char * new_name = new char[name_len];
-                            strncpy(new_name, section_search_start, name_len);
-                            replace_special_characters(new_name, name_len);
-                            this->coffStream->seek(0);
-                            this->coffStream->seek(section_data_start_offset + section_data_offset);
-                            this->coffStream->write(new_name, name_len);
-                            delete new_name;   
-                            section_search_start += name_len+1;                   
-                        }
-                    }
-                    // search section data for extensionless name
-                    name_len = name_no_ext.size();
-                    section_search_start = &*section;
-                    while(section_search_start && (section_search_start < search_terminator)) {
-                        section_search_start = findstr(section_search_start, name_no_ext.c_str(), data_size);
-                        if (section_search_start) {
-                            // we found a name, rename
-                            ptrdiff_t offset = section_search_start - section;
-                            char * new_name = new char[name_len];
-                            strncpy(new_name, section_search_start, name_len);
-                            replace_special_characters(new_name, name_len);
-                            this->coffStream->seek(0);
-                            this->coffStream->seek(section_data_start_offset + offset);
-                            this->coffStream->write(new_name, name_len);    
-                            delete new_name;
-                            section_search_start += name_len+1;                
-                        }
-                    }
+                    this->NormalizeSectionNames(name_no_ext, section, section_data_start_offset, data_size);
```

Comment:
Should this also be called on `name`?

Reply:
It occured to me that the name without the extension is a substring of the name with the extension, and as this method parses the entire section looking for a name regardless of how many names it finds, everything will get renamed regardless.

Reply:
Last question: if you encounter `name_with_extension`, what will the result look like when this overwrites? I assume this null-terminates name, so would that truncate off the extension in cases where the extension appears?

Reply:
It does not null terminate the name. The COFF spec does not indicate that the name should be null terminated.